docs: sync visualization docs with source (#126809)
grafana/grafana: docs/sources/visualizations/panels-visualizations/visualizations/bar-chart/index.md

## Apply filters from the bar chart

In bar charts, you can apply filters directly from the visualization.
Filters are available when the x-axis field is filterable and the data source supports dashboard drilldown filters, such as Prometheus or Loki.

To display the filter button, hover your cursor over the bar that has the value for which you want to filter and click the bar:

{{< figure src="/media/docs/grafana/panels-visualizations/screenshot-adhoc-filter-icon-bar-v12.2.png" max-width="300px" alt="The filter button in a bar chart tooltip">}}

> For more information about applying filters this way, refer to [Dashboard drilldown with filters](http://grafana.com/docs/grafana/<GRAFANA_VERSION>/visualizations/dashboards/variables/add-template-variables/#dashboard-drilldown-with-filters).

Author: ```suggestion
For more information about applying filters this way, refer to [Dashboard drilldown with filters](https://grafana.com/docs/grafana/<GRAFANA_VERSION>/visualizations/dashboards/build-dashboards/filter-group-by/#dashboard-drilldown-with-filters).
```

| X Axis                           | Specify which field is used for the x-axis.          |
| Orientation                      | Choose from: <ul><li>**Auto** - Grafana decides the bar orientation based on the panel dimensions.</li><li>**Horizontal** - Will make the X axis the category axis.</li><li>**Vertical** - Will make the Y axis the category axis.</li></ul> |
| Rotate x-axis tick labels        | When the graph is vertically oriented, this setting rotates the labels under the bars. This setting is useful when bar chart labels are long and overlap.  |
| X-axis tick label max length | Sets the maximum length of bar chart labels. Labels longer than the maximum length are truncated, and appended with `...`. This option is available when **Rotate x-axis tick labels** isn't `0`. |

> | X-axis labels minimum spacing | Sets the minimum spacing between x-axis labels. Depending on your choice, you can select the **RTL** checkbox to require space from the right side. Choose from: <ul><li>**None** - All tick marks are shown.</li><li>**Small** - 100 px of space is required between labels.</li><li>**Medium** - 200 px of space is required between labels.</li><li>**Large** - 300 px of space is required between labels.</li></ul>  |

Author: ```suggestion
| X-axis labels minimum spacing | Sets the minimum spacing between x-axis labels. Depending on your choice, you can select the **RTL** checkbox to require space from the right side. Choose from: <ul><li>**None** - All tick marks are shown.</li><li>**Small** - 100 pixels of space is required between labels.</li><li>**Medium** - 200 pixels of space is required between labels.</li><li>**Large** - 300 pixels of space is required between labels.</li></ul>  |
```

| Bar radius                       | Controls the radius of the bars. Choose from: <ul><li>0 = Minimum radius</li><li>0.5 = Maximum radius</li></ul>     |
| Highlight full area on hover     | Controls if the entire surrounding area of the bar is highlighted when you hover over the bar. This option is available only when **Stacking** is **Off**.    |
| Color by field    | Use the color value for a sibling field to color each bar value. |
| Line width                       | Controls line width of the bars. This is a field option and can be set in field defaults or overrides.    |
| Fill opacity                     | Controls the fill opacity of bars. This is a field option and can be set in field defaults or overrides.                    |
| [Gradient mode](#gradient-mode)  | Set the mode of the gradient fill. Fill gradient is based on the line color. To change the color, use the standard color scheme field option. Gradient appearance is influenced by the **Fill opacity** setting. This is a field option and can be set in field defaults or overrides.                                                                               |

> | Transform                        | Apply an override-only transform to a field. Choose **Constant** to show the first value as a constant bar, or **Negative Y** to flip values to the negative y-axis. |

Author: ```suggestion
```
